typescript/src/services/redemptions/redemptions-service.ts

```diff
@@ -53,19 +58,46 @@ export class RedemptionsService {
     targetChainTxHash: Hex
     walletPublicKey: Hex
   }> {
-    const { walletPublicKey, mainUtxo, redeemerOutputScript } =
-      await this.determineRedemptionData(bitcoinRedeemerAddress, amount)
+    try {
+      const candidateWallets = await this.fetchWalletsForRedemption()
+      const { walletPublicKey, mainUtxo, redeemerOutputScript } =
+        await this.determineValidRedemptionWallet(
+          bitcoinRedeemerAddress,
+          amountToSatoshi(amount),
+          candidateWallets
+        )
 
-    const txHash = await this.tbtcContracts.tbtcToken.requestRedemption(
-      walletPublicKey,
-      mainUtxo,
-      redeemerOutputScript,
-      amount
-    )
+      const txHash = await this.tbtcContracts.tbtcToken.requestRedemption(
+        walletPublicKey,
+        mainUtxo,
+        redeemerOutputScript,
+        amount
+      )
 
-    return {
-      targetChainTxHash: txHash,
-      walletPublicKey,
+      return {
+        targetChainTxHash: txHash,
+        walletPublicKey: walletPublicKey,
+      }
+    } catch (error) {
+      console.error(
```

Comment:
```suggestion
      console.warn(
```

Reply:
Addressed on [chore: update console in redemption service](https://github.com/keep-network/tbtc-v2/pull/850/commits/4c6c5ffad7ffaa66bb4148118f6c0c7029faa433)